agent/lib/kontena/observable.rb

```diff
@@ -1,75 +1,209 @@
+  # If the observable already has a value, then observing that value will return/yield the immediate value.
+  # If the observable crashes, then any Observers will immediately raise.
+  #
+  # TODO: are you allowed to reset an observable after crashing it, allowing observers to restart and re-observe?
+  #
+  # @attr observers [Hash{Kontena::Observer => Boolean}] stored value is the persistent? flag
+  class Observable
+    require_relative './observable/registry'
+
+    # @return [Celluloid::Proxy::Cell<Kontena::Observable::Registry>] system registry actor
+    def self.registry
+      Celluloid::Actor[:observable_registry] || fail(Celluloid::DeadActorError, "Observable registry actor not running")
+    end
+
     include Kontena::Logging
 
-    def observable_value
-      @observable_value
+    attr_reader :logging_prefix # customize Kontena::Logging#logging_prefix by instance
+
+    class Message
+      attr_reader :observer, :observable, :value
+
+      # @param observer [Kontena::Observer]
+      # @param observable [Kontena::Observable]
+      # @param value [Object, nil, Exception]
+      def initialize(observer, observable, value)
+        @observer = observer
+        @observable = observable
+        @value = value
+      end
+    end
+
+    # mixin for Celluloid actor classes
+    module Helper
+      # Create a new Observable using the including class name as the subject.
+      # Register the Observable with the Kontena::Observable::Registry.
+      # Links to the registry to crash the Observable if the owning actor crashes.
+      #
+      # @return [Kontena::Observable]
+      def observable
+        return @observable if @observable
+
+        # the register can suspend this task, so other calls might get the observable before it gets registered
+        # shouldn't be a problem, unless the register/linking somehow fails and crashes this actor without crashing the
+        # observable?
+        @observable = Kontena::Observable.new(self.class.name)
+
+        observable_registry = Kontena::Observable.registry
+        observable_registry.register(@observable, self.current_actor)
+
+        self.links << observable_registry # registry monitors owner
+
+        @observable
+      end
+    end
+
+    # @param subject [Object] used to identify the Observable for logging purposes
+    def initialize(subject = nil)
+      @subject = subject
+      @mutex = Thread::Mutex.new
+      @observers = {}
+      @value = nil
+
+      # include the subject (owning actor class, other resource) in log messages
+      @logging_prefix = "#{self}"
+    end
+
+    # @return [String]
+    def to_s
+      "#{self.class.name}<#{@subject}>"
+    end
+
+    # @return [Object, nil] last updated value, or nil if not observable?
+    def get
+      @value
+    end
+
+    # Observable has updated, and has not reset. It might be crashed?
+    #
+    # @return [Boolean]
+    def observable?
```

Comment:
Rename this to `ready?` - that's the term used elsewhere, and `observable?` doesn't really make any sense here anymore.